mlflow/pyfunc/model.py

```diff
@@ -937,29 +943,63 @@ def create_text_delta(self, delta: str, item_id: str) -> dict[str, Any]:
                 "delta": delta,
             }
 
-        def create_text_output_item(self, text: str, id: str) -> dict[str, Any]:
+        @staticmethod
+        def create_annotation_added(
+            item_id: str, annotation: dict[str, Any], annotation_index: int | None = 0
+        ) -> dict[str, Any]:
+            return {
+                "type": "response.output_text.annotation.added",
+                "item_id": item_id,
+                "annotation_index": annotation_index,
+                "annotation": annotation,
+            }
+
+        @staticmethod
+        def create_text_output_item(
+            text: str, id: str, annotations: list[dict[str, Any]] | None = None
+        ) -> dict[str, Any]:
             """Helper method to create a dictionary conforming to the text output item schema.
 
             Read more at https://mlflow.org/docs/latest/genai/flavors/responses-agent-intro#creating-agent-output.
 
             Args:
                 text (str): The text to be outputted.
                 id (str): The id of the output item.
+                annotations (Optional[list[dict]]): The annotations of the output item.
             """
+            content_item = {
+                "text": text,
+                "type": "output_text",
+            }
+            if annotations is not None:
+                content_item["annotations"] = annotations
             return {
                 "id": id,
-                "content": [
+                "content": [content_item],
+                "role": "assistant",
+                "type": "message",
+            }
+
+        @staticmethod
+        def create_reasoning_item(id: str, reasoning_text: str) -> dict[str, Any]:
```

Comment:
```suggestion
        def _create_reasoning_item(id: str, reasoning_text: str) -> dict[str, Any]:
```

should this be private?

Reply:
same comment on other helper methods.